x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -310,9 +302,10 @@ protected boolean skipResolved() {
         }
     }
 
-    private class ResolveTable extends AnalyzerRule<UnresolvedRelation> {
-        @Override
-        protected LogicalPlan rule(UnresolvedRelation plan) {
+    private static class ResolveTable extends ParameterizedAnalyzerRule<UnresolvedRelation, AnalyzerContext> {
+
+        protected LogicalPlan rule(UnresolvedRelation plan, AnalyzerContext context) {
+            var indexResolution = context.indexResolution();
```

Comment:
I would have kept the explicit type of the variable - `IndexResolution`.

Reply:
I reverted though I personally find it redundant - `IndexResolution indexResolution = context.indexResolution()`